frame/membership/src/migrations/v4.rs

```diff
@@ -119,26 +112,24 @@ pub fn post_migrate<P: GetStorageVersion, N: AsRef<str>>(old_pallet_name: N, new
 	let new_pallet_name = new_pallet_name.as_ref();
 	log_migration("post-migration", old_pallet_name, new_pallet_name);
 
+	// Assert that nothing remains at the old prefix.
 	let old_pallet_prefix = twox_128(old_pallet_name.as_bytes());
-	#[cfg(test)]
-	{
-		let storage_version_key =
-			[&old_pallet_prefix, &twox_128(STORAGE_VERSION_STORAGE_KEY_POSTFIX)[..]].concat();
-		assert!(storage::next_key(&old_pallet_prefix)
-			.map_or(true, |next_key| !next_key.starts_with(&old_pallet_prefix) ||
-				next_key == storage_version_key));
-	}
-	#[cfg(not(test))]
-	{
-		// Assert that nothing remains at the old prefix
-		assert!(storage::next_key(&old_pallet_prefix)
-			.map_or(true, |next_key| !next_key.starts_with(&old_pallet_prefix)));
-	}
+	let old_pallet_prefix_iter = frame_support::storage::KeyPrefixIterator::new(
+		old_pallet_prefix.to_vec(),
+		old_pallet_prefix.to_vec(),
+		|_| Ok(()),
+	);
+	assert_eq!(old_pallet_prefix_iter.count(), 0);
 
+	// Assert that at least one key have been moved to the new prefix.
+	// NOTE: storage_version_key is already in the new prefix.
 	let new_pallet_prefix = twox_128(new_pallet_name.as_bytes());
-	// Assert that the storages have been moved to the new prefix
-	assert!(storage::next_key(&new_pallet_prefix)
-		.map_or(true, |next_key| next_key.starts_with(&new_pallet_prefix)));
+	let new_pallet_prefix_iter = frame_support::storage::KeyPrefixIterator::new(
+		new_pallet_prefix.to_vec(),
+		new_pallet_prefix.to_vec(),
+		|_| Ok(()),
+	);
+	assert!(new_pallet_prefix_iter.count() > 1);
```

Comment:
in the past we ensure a key was at the new prefix, considering the storage_version_key is expected there, it wasn't checking much. Now we ensure at least another one key is there.
Actually maybe membership is in a state when nothing is stored. in this case the post_migration gives a false positive